intro-to-oop-contd/projects-with-many-files.md

```diff
@@ -264,79 +501,301 @@ Where `passenger.py` looks like:
+    │   └── front_camera.py
+    └── tests
+        ├── __init__.py
+        ├── processor_test.py
+        ├── filters
+        │   ├── __init__.py
+        │   ├── cat_ears_filter_test.py
+        │   └── sepia_filter_test.py
+        └── image_source
+            ├── __init__.py
+            ├── back_camera_test.py
+            └── front_camera_test.py
+```
+* ```
+photo-app/
+├── README.md
+├── requirements.txt
+├── photo_app
+│   ├── __init__.py
+│   ├── __main__.py
+│   ├── back_camera.py
+│   ├── cat_ears_filter.py
+│   ├── front_camera.py
+│   ├── processor.py
+│   └── sepia_filter.py
+└── tests
+    ├── __init__.py
+    ├── back_camera_test.py
+    ├── cat_ears_filter_test.py
+    ├── front_camera_test.py
+    ├── processor_test.py
+    └── sepia_filter_test.py
+```
+
+##### !end-options
+
+##### !answer
+
+* ```
+photo-app/
+├── README.md
+├── requirements.txt
+├── photo_app
+│   ├── __init__.py
+│   ├── __main__.py
+│   ├── processor.py
+│   ├── filters
+│   │   ├── __init__.py
+│   │   ├── cat_ears_filter.py
+│   │   └── sepia_filter.py
+│   └── image_source
+│       ├── __init__.py
+│       ├── back_camera.py
+│       └── front_camera.py
+└── tests
+    ├── __init__.py
+    ├── processor_test.py
+    ├── filters
+    │   ├── __init__.py
+    │   ├── cat_ears_filter_test.py
+    │   └── sepia_filter_test.py
+    └── image_source
+        ├── __init__.py
+        ├── back_camera_test.py
+        └── front_camera_test.py
+```
+
+##### !end-answer
+
+##### !explanation
+
+A good default layout is to have our general project files in the project root. Then we create two top-level packages, one for our application code, and one for the tests. The structures under each should generally mirror one another.  Each package folder should have an `__init__.py` file. And when we start dealing with a number of files, some of which are more closely related to each other than the rest of the package, we should consider creating subfolders to group the related files.
+
+##### !end-explanation
+
+### !end-challenge
+<!-- prettier-ignore-end -->
+<!-- ======================= END CHALLENGE ======================= -->
 
 <!-- prettier-ignore-start -->
 ### !challenge
 
 * type: multiple-choice
 * id: f032adcb-c05b-4591-96b9-58bc24d7ee79
-* title: Importing modules
+* title: Projects With Many Files
 
 ##### !question
 
-Hosts for AdaBnB can host guests at multiple properties on the app. Each property is an instance of a `Place` class. If we ran `guest.py`, we would receive a `NameError`. Select the option that describes how to solve this `NameError`.
+Hosts for AdaBnB can host guests at multiple properties on the app. Each property is an instance of a `Place` class. If we tried to call the `create_places` method of `Host`, we would receive a `NameError`. Assuming the `place` module in which the `Place` class is defined is in the same package folder as `host.py`, select the option that best describes how to solve this `NameError`.
```

Comment:
Soft suggestion: The sentence is good, but it might be nice to give them a diagram either additonally, or instead.

For convenience

```
adabnb/
├── README.md
├── requirements.txt
├── adabnb
│   ├── __init__.py
│   ├── host.py
│   └── place.py
└── tests
    ├── __init__.py
    ├── host_test.py
    └── place_test.py
```

Didn't make a suggestion _because code formatting_.

Reply:
Good idea! And thanks for the tree!